src/App.jsx

```diff
@@ -1,35 +1,65 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import { useState, useEffect } from "react";
+import axios from "axios";
+import CharacterList from "./components/CharacterList";
+import "./App.module.css";
 
 function App() {
-  const [count, setCount] = useState(0)
+  const [characters, setCharacters] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(null);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [hasMore, setHasMore] = useState(true);
+
+  const fetchCharacters = async (page = 1) => {
+    setIsLoading(true);
+    try {
+      const response = await axios.get(
+        `https://rickandmortyapi.com/api/character?page=${page}`
+      );
+
+      setCharacters((prevCharacters) => {
+        const newCharacters = response.data.results.filter(
+          (character) =>
+            !prevCharacters.some((prev) => prev.id === character.id)
+        );
+        return [...prevCharacters, ...newCharacters];
+      });
+
+      setHasMore(response.data.info.next !== null);
+      setIsLoading(false);
+    } catch (err) {
+      setError(err);
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    fetchCharacters(currentPage);
+  }, [currentPage]);
+
+  useEffect(() => {
+    const handleScroll = () => {
+      if (
+        window.innerHeight + document.documentElement.scrollTop >=
+          document.documentElement.offsetHeight - 100 &&
+        hasMore &&
+        !isLoading
+      ) {
+        setCurrentPage((prevPage) => prevPage + 1);
+      }
+    };
+
+    window.addEventListener("scroll", handleScroll);
+    return () => window.removeEventListener("scroll", handleScroll);
```

Comment:
read about best practices of infinite scroll implementation, scroll event triggers to often and this is not good 

HINT: you have to use some web api